app/controllers/home_controller.rb

```diff
@@ -94,21 +94,39 @@ def activity
       .where(type: 'note', status: 1)
       .where('term_data.name = (?)', 'hidden:response')
       .collect(&:nid)
-    notes = Node.where(type: 'note')
+
+    basenotes = Node.where(type: 'note')
       .where('node.nid NOT IN (?)', hidden_nids + [0]) # in case hidden_nids is empty
       .order('nid DESC')
-      .page(params[:page])
-    notes = notes.where('nid != (?)', blog.nid) if blog
+    basenotes = basenotes.where('nid != (?)', blog.nid) if blog
+    notes = basenotes
+
+    questions = Tag.find_nodes_by_type('question:question', 'note', 999_999_999_999_999)
```

Comment:
Here you should be able to pass `nil` and the limit will be lifted. But unfortunately, this will query the database for all `question` nodes, and likewise below all `event` nodes and above all `notes` -- on the production server this will be thousands of records, so we need to optimize for it to work. The `.page()` method is quite sophisticated, and uses `limit()` and `offset()` to select just the right records to display on the given page. But the problem we're encountering here is that we are trying to subselect across different tables. I don't believe the `will_paginate` gem can do this. 

What we could do here is to start by fetching the paginated notes, by leaving in `.page()` for that query. For events and questions, which are also nodes, we could try to filter them in/out if they are not selected... this could be difficult but we could re-use some of the code in Tag.find_nodes_by_type, maybe... 

Then, we could use the timestamps from the nodes we fetched to select the comments which were made during the same period. These could be added in, and while the calculation of total # of pages would be correct, the number of items per page would vary since we would be calculating # of pages based only on nodes. 

I'm trying to think of how this could be done more simply but it was already a pretty complex thing to display in the first place... the critical part now is to not return the entire table's worth of each type on each page load. This works on a very small dataset but would almost certainly crash the server when 1000s of records are returned each time in production.

Reply:
https://stackoverflow.com/a/7959348 suggests synchronizing the `offset`, but then over time the comments and nodes would drift because they don't occur at the same rate over time.

Reply:
OK, here's an idea. I don't know if it's good, but just to think through it...

Let's simplify - let's put wiki edits permanently in the sidebar and display them below in mobile view, so we just don't worry about wikis as part of this. Then we're left with Notes, Events, Questions (all types of Node) and Comments.

The first three we could probably filter if we wanted based on what tags they have, using joins.  But Comments messes us up, because it's a different table (see my above comment). What if we, instead of showing pure comments, showed only Nodes in Activity, but we joined comments where they exist (let's forget Answer comments for now) and we sorted by `comment.timestamp` and then `node.created`: https://api.rubyonrails.org/classes/ActiveRecord/QueryMethods.html#method-i-order -- would that work? And then in the views, we detect if there is a comment attached and we display the `X commented on Y Z time ago` instead of the usual note template? 

This isn't air-tight, but seems plausible. I'm sorry I missed the change in pagination from the very first commit. I'm going to think on this more and happy to hear your thoughts too.

Reply:
In the above scenario, the right type of join will return multiple duplicate Node records, but one for each comment, and ordered by comment timestamp. I think that's fine, because each would really be a "stand in" for the associated comment record. But alternatively we could try to use a join type which would return only unique/discrete Node records, but choose the most recent associated comment record. Then we could show the Node template on the dashboard, and kind of "attach" the most recent comment to the bottom of it, so as to show why it's appearing in the timeline at that point. Just thinking through possibilities here.

Reply:
Finally, because this is already quite complex, we could stop offering to filter by Event and Question, simply grouping all Nodes together. That would simplify things a good bit, leaving only All, "Posts" (we'll call them?), and Comments. Trying to think of how to make this a reasonable path forward.

Reply:
You know what? I have a simpler idea. I know it's not that great but it's better than fetching thousands. We could just fetch the first 37 from all of the tables, because it's the nodes limit for one page. You know what I mean? It's a temporary solution, but it's not that bad and it could be solved in another PR.

Reply:
I think this is doable, and I agree with the principle of doing something now and following up in later PRs. But this would start to go out of time sync pretty fast because there are more comments per time period than nodes. Could we try the simpler solution but using timestamps instead of record count?

Reply:
So you'd narrow your query with a `where()` and I think you can even pass a range to `where()` in activerecord... then it'd be like `.where(timestamp: basenotes.first.created..basenotes.last.created)` i think?